src/documents/PromisesAPI.md

```diff
@@ -0,0 +1,236 @@
+code, and so we only have limited time to spend thinking about how to make the code we do have conform to a faulty ideal of what we dreamed
+that the virtues of conventional, synchronous code might once have been. However, it's worth noting that there is at least [one virtue](#promise-accumulaterejectionreason-error-) of 
+conventional, synchronous code that is recaptured by no other promise system but ours. Also, this [github issue](https://github.com/promises-aplus/promises-spec/issues/4) attached to the 
+A+ promises specification is a useful source of convincing argumentation that the decision in favour of universally asynchronous resolution is flawed.
+
+As a further landmark for discussion, note that in terms of the following very illuminating [category theoretic](http://brianmckenna.org/blog/category_theory_promisesaplus) treatment of 
+promises (itself rejected by mainstream promises proponents), our `then` method is very definitely ***not*** "the name for `flatMap`". 
+Our `then` method is simply a "glorified callback aggregator".
+
+The implementation skeleton for Infusion's promises was taken from a code sample by John Hann (unscriptable) in this [gist](https://gist.github.com/unscriptable/814052) - full credit and thanks,
+and please read the gist for some further commentary and coverage of limitations.
+
+
+## Core Promises Utility API
+
+The library implements a few utilities without which it is inconvenient to use promises:
+
+### fluid.isPromise(totest)
+
+* `totest {Any}` An object to be checked for being a promise
+* Returns: `{Boolean}` If `totest` has a member `then` of type `Function`, returns `true`.
+
+Determines whether an object is a promise, for our purposes. Any object with a member `then` of type `Function` passes this test.
+This includes essentially every known variety, including jQuery promises. This test in fact identifies what in other libraries is
+termed a "foreign thenable".
+
+### fluid.toPromise(value)
+
+* `value {Any}` A value to be converted ("hoisted") to a promise
+* Returns: `{Promise}` If the supplied value is already a promise, it is returned unchanged. Otherwise a fresh promise is created with the value as resolution and returned.
+
+Coerces any value to a promise. If it is already a promise, it is returned unchanged. Otherwise, `toPromise` constructs a fresh promise resolving to the supplied value.
+
+
+### fluid.promise.follow(source, target)
+
+* `source {Promise}` A promise which is to be followed in its resolution.
+* `target {Promise}` A promise which will follow the `source` in its resolution. `target` will receive a call to `then` causing it to resolve when `source` resolves,
+and reject when `source` rejects. 
+
+Chains the resolution methods of one promise (target) so that they follow those of another (source).
+That is, whenever source resolves, target will resolve, or when source rejects, target will reject, with the
+same payloads in each case.
+
+### fluid.promise.map(source, func)
```

Comment:
This "mapping" is not very well explained here. Mapped to what? Could you elaborate a bit?